flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -279,6 +282,10 @@ public void start() throws Exception {
                     miniClusterConfiguration.getRpcServiceSharing() == RpcServiceSharing.SHARED;
 
             try {
+                final PluginManager pluginManager =
+                        PluginUtils.createPluginManagerFromRootFolder(configuration);
+                FileSystem.initialize(configuration, pluginManager);
+
```

Comment:
Is `MiniCluster` always and only used for `LocalExecutionEnvironment`? I think the fix is legit but I'm wondering if the description is accurate. Are we solving an issue in the `LocalExecutor` or in `MiniCluster`?
I also don't quite understand why `ClusterEntrypoint` is not used at all in `LocalExecutionEnvironment`. @zentol could you please also have peek?

Reply:
> Is MiniCluster always and only used for LocalExecutionEnvironment?

No. It can also be used if the Test(Stream)Environments are explicitly set by the user.

Reply:
The ClusterEntryPoint is used for actual clusters where the TMs run externally. The MiniCluster is the single-JVM equivalent.